app/src/main/java/com/example/sabonit/Account.java

```diff
@@ -0,0 +1,120 @@
+    String phoneNumber;
+    // User's cart, empty when initialized
+    Cart cart;
+    // User's history, empty when initialized
+    History history;
+
+
+    /* ********* Constructors: ********* */
+    /**
+     * Default constructor.
+     */
+    public Account() {
+    }
+
+    /**
+     * When signing up, the account is initialized for the user's sata handling.
+     * @param location - the address that the Sabonit get in the order.
+     * @param name - user's name.
+     * @param phoneNumber - user's phone number.
+     */
+    public Account(String location, String name, String phoneNumber) {
+        this.location = location;
+        this.name = name;
+        this.phoneNumber = phoneNumber;
+        this.cart = new Cart();
+        this.history = new History();
+    }
+
+    /* ********* Getters & Setters: ********* */
+
+    public String getLocation() {
+        return location;
+    }
+
+    public void setLocation(String location) {
+        this.location = location;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String getPhoneNumber() {
+        return phoneNumber;
+    }
+
+    public void setPhoneNumber(String phoneNumber) {
+        this.phoneNumber = phoneNumber;
+    }
+
+    public Cart getCart() {
+        return cart;
+    }
+
+    public History getHistory() {
+        return history;
+    }
+}
+
+/* // todo - move the next code to the appropriate activity: responsible of extracting relevant data
+             of the user and check validity of it.
+
+        this.phoneNumber = getValidPhoneNumber(phoneNumber);
+        // todo add a unique check
+        if (!isPhoneNumberValid(phoneNumber))
+        {
+            TelephonyManager tMgr = (TelephonyManager)mAppContext.getSystemService(Context.TELEPHONY_SERVICE);
+            String mPhoneNumber = tMgr.getLine1Number();
+        } else
+            this.phoneNumber = phoneNumber;
+
+    private String getValidPhoneNumber(String phoneNumber, Context appContext)
+    {
+        // validity check of input - checks if phoneNumber is a String
+        TelephonyManager telephonyManager;
+        if (!(phoneNumber instanceof String))
+            telephonyManager = (TelephonyManager)appContext.getSystemService(Context.TELEPHONY_SERVICE);
+            return telephonyManager.getLine1Number();
+        // remove all white spaces and all dashes (and anything that is not a digit)
+        phoneNumber = phoneNumber.replaceAll("\\D", "");
+        // First condition - for numbers in the shape "0_-___-____"
+        // Second condition - for numbers in the shape "+972-__-___-____"
+        if (phoneNumber.length() == IMPLICIT_ISRAELI_PHONE_NUMBER_LENGTH && phoneNumber.charAt(0) == 0 ||
+            phoneNumber.length() == ISRAELI_AREA_CODE && phoneNumber.substring(0, 4).equals(ISRAELI_AREA_CODE_INFIX))
+            return phoneNumber;
+    }
+
+ */
```

Comment:
needs to think about how to implement it

Reply:
Yeah, it's a big issue... we need to think about it together.
To cut a long story short, this piece of code handles in a generic way in the validity of the phone number, there are some versions of it in the internet and i implemented a new version by myself